pkg/server/nodemanager/node_controller.go

```diff
@@ -330,6 +330,7 @@ func (c *NodeController) stopSingleNode(node *api.Node) error {
 	}
 	c.NodeClientFactory.DeleteClient(node.Status.Addresses)
 	go func(n *api.Node) {
+		_ = c.CloudClient.RemoveRoute("", n.Status.InstanceID)
```

Comment:
> In the current implementation, this results in a noop in the cloudClient (and returns an error)? Seems like the check in AWS is if cidr == "" || instanceID == "" { return fmt.Errorf(...)}

Thanks, that was a bug. In RemoveRoute(), we need at least one input parameter (either the instance ID or the CIDR, and RemoveRoute() will match for one if only one of them is non-empty, otherwise for both), in AddRoute() we need both.

> I'm unsure if this is needed here since we're also removing the route in the pod controller. 

I added it to both places, since there are edge cases around Kip getting restarted when the pod is being terminated. I think 99.99% of the time it is unnecessary to call RemoveRoute() from both places.

Let me remove the second call, and instead clean up dangling instance routes via the garbage controller. We can simply check blackhole instance routes and remove them there instead of the extra RemoveRoute() call from the node controller.

Reply:
Cleaning up blackhole routes sounds good.